Source/pfile.cpp

```diff
@@ -34,13 +34,17 @@ namespace {
 /** List of character names for the character selection screen. */
 char hero_names[MAX_CHARACTERS][PLR_NAME_LEN];
 
+std::string savePrefix;
+
 std::string GetSavePath(uint32_t saveNum)
 {
 	std::string path = paths::PrefPath();
 	const char *ext = ".sv";
 	if (gbIsHellfire)
 		ext = ".hsv";
 
+	path.append(savePrefix);

```

Comment:
This is a bit of a hack. If instead we change it so that the demo file is independent of the original save then we could create a normal save game and compare that to our known hash with out it affecting the demo (I presume avoiding overwriting the demo-start-point-save-game is the reason for this).

Reply:
Your assumption is correct. 🙂 
But if you write a new save game, you aren't allowed to override the existing (regardless if this is the original or the demo starting save game or if you create the savegame only for hashcode generation). So I think this topic stays or I miss something. 😉 
If you want I can take a look to change `pfile_write_hero` so that it get's the filename passed in. Additional question: if yes in this pr or new pr? 😉

Reply:
If you think you will do that in the next PR I'm ok merging as is :)

Reply:
I will check it in a next pr.
If it's too much work (rabbit hole) I will open a pr that adds `BUILD_TESTING` defines.